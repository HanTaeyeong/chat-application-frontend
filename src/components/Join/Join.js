import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

import './Join.scss'

export default function Join() {
  const [inputValue, setInputValue] = useState({ name: "", room: '' })
  const { name, room } = inputValue;
  const navigate = useNavigate();

  const onChange = (e) => setInputValue({ ...inputValue, [e.target.name]: e.target.value })

  const onClick = (e) => {
    if (!name || !room) { return }
    e.preventDefault();

    const targetPath = `/chat?name=${name}&room=${room}`;
    navigate(targetPath)
  }

  return <div id='join-outer-container'>
    <div className='join-inner-container'>
      <h1 id='join-heading'>Join</h1>

      <input id='#join-input-name' className='join-input' placeholder='Name' name='name'
        onChange={onChange} onKeyPress={e => e.key === "Enter" && onClick(e)} />
      <input id='#join-input-room' className='join-input' placeholder='Room' name="room"
        onChange={onChange} onKeyPress={e => e.key === "Enter" && onClick(e)} />

      <div onClick={onClick}  >
        <button className='join-sign-in-button' >Sign In</button>
      </div>

    </div>
  </div>;
}
