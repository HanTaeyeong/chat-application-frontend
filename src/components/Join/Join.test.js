import Join from './Join'



describe('Join Component Test',()=>{
    
    test('render test',()=>{})

})