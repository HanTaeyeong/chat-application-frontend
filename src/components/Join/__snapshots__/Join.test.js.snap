// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Component Test render test 1`] = `
<div
  id="join-outer-container"
>
  <div
    className="join-inner-container"
  >
    <h1
      id="join-heading"
    >
      Join
    </h1>
    <input
      className="join-input"
      id="#join-input-name"
      name="name"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Name"
    />
    <input
      className="join-input"
      id="#join-input-room"
      name="room"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Room"
    />
    <div
      onClick={[Function]}
    >
      <button
        className="join-sign-in-button"
      >
        Sign In
      </button>
    </div>
  </div>
</div>
`;
